### ISSUE TITLE
Unlock with Apple Watch

### ISSUE THREAD

**Issue by david-shortman** [buttercup/buttercup-desktop#961]
1Password recently introduced a feature to unlock vaults with Apple Watch authentication (which must grant access to the master password in the Keychain to the application).

Are OS-specific features welcome in buttercup-desktop?

**Comment by perry-mitchell** [comment]
@david-shortman Yes, I'd say that they are, as long as we're strict on stability with regards to non-mac systems (with this specific feature). I like this idea. There's a couple of ideas to explore:

 * This would work on _any_ platform if the vault is using My Buttercup as a source (potentially)
 * This could work using Apple's built-in systems (icloud?) - not sure how the watch would communicate this with the Mac, but it'd need to have the vault password stored encrypted somewhere.

**Comment by david-shortman** [comment]
I think more broadly macOS support would include the ability to access a vault using TouchID as well. The Unlock with Watch API appears to be used in the same vein as unlocking with Touch ID, in that you're granting access to an item in the Keychain.

**Comment by david-shortman** [comment]
Looks like this is what allows using bio or watch auth- https://developer.apple.com/documentation/localauthentication/klapolicydeviceownerauthenticationwithbiometricsorwatch?language=objc
